### DISCUSSION TITLE
"TypeError: Cannot read property 'range' of null" with async keyword

### DISCUSSION THREAD

**Discussion by Akshitoo8** [eslint/eslint#17307] [Q&A]
I am successfully using ESLINT in an old project for about a year. 

To add a delay in a method, I used `await delay(1500)` for which I had to make the method `async`.
As soon as I introduce async keyword, it gives me the following error:

> TypeError: Cannot read property 'range' of null
>     at TokenStore.getTokenBefore (project/node_modules/eslint/lib/token-store/index.js:318:18)
>     at EventEmitter.checkFunction (project/node_modules/eslint/lib/rules/generator-star-spacing.js:131:42)
>     at EventEmitter.emit (events.js:326:22)
>     at NodeEventGenerator.applySelector (project/node_modules/eslint/lib/util/node-event-generator.js:265:26)

If I add the following rules to eslintrc as suggested by the community, I get an error.
```
    "template-curly-spacing" : "off",
    "indent": ["error", 2, {
      "ignoredNodes": ["TemplateLiteral"]
    }],
```
Error: 	
> Configuration for rule "indent" is invalid:
> 	Value "data["1"].ignoredNodes" has additional properties.

I am using the following versions:

> 		"eslint": "^3.6.0",
> 		"eslint-config-airbnb": "^12.0.0",
> 		"eslint-import-resolver-webpack": "^0.8.0",
> 		"eslint-plugin-import": "^2.2.0",

>		node:  v12.22.12 
>		npm 6.14.16

The process uses `npm run lint` to check the lint errors.

**Comment by Rec0iL99** [comment]
Hi @Akshitoo8, it looks like you are using an old version of ESLint (`3.6.0`). Can you please try upgrading to a newer version?

**Comment by aladdin-add** [comment]
https://github.com/babel/babel-eslint/issues/799#issuecomment-570790316

**Comment by sam3k** [comment]
@Akshitoo8 was the information provided above helpful?

**Comment by Akshitoo8** [comment]
Thanks for the suggestions but unfortunately I couldn't try them. 
I was in a hurry and resolved the error by changing the 'function' to a 'const'
`async function methodName(a, b) {}`
to
`const methodName = async (a, b) => {}`
